src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -251,6 +267,7 @@ protected override void Dispose(bool disposing)
                 // Prepare context objects
                 state.ResponseMessage = new CurlResponseMessage(state.RequestMessage);
                 stateHandle = GCHandle.Alloc(state);
+                state.Handler = this;
```

Comment:
Can this assignment be done when the RequestCompletionSource is created?  If possible, it'd be even nicer to make the associated handler a readonly field on the RequestCompletionSource and pass it to its ctor, in order to make it very clear that the RequestCompletionSource is directly tied to a handler that'll never change.